### PR TITLE
add support for logging to AWS CloudWatch Logs

### PR DESCRIPTION
**Pull request by tobiasmcnulty**

### CODE REVIEW

File: fabulaws/library/wsgiautoscale/api.py [hunk 1]
```diff
@@ -187,12 +187,32 @@ def _setup_env(deployment_tag=None, environment=None, override_servers={}):
...
+        ('postgresql', '/var/log/postgresql/postgresql-*.log', '%Y-%m-%d %H:%M:%S'),
+        ('redis', '/var/log/redis/redis*.log', '%d %b %H:%M:%S'),
+        ('rabbitmq', '/var/log/rabbitmq/rabbit*.log', '%d-%b-%Y::%H:%M:%S'),
+        ('nginx', os.path.join(env.log_dir, 'error.log'), '%Y/%m/%d %H:%M:%S'),
+        ('gunicorn', os.path.join(env.log_dir, 'gunicorn.log'), '%Y-%m-%d %H:%M:%S'),
+        ('pgbouncer', os.path.join(env.log_dir, 'pgbouncer.log'), '%Y-%m-%d %H:%M:%S'),
+        ('stunnel', os.path.join(env.log_dir, 'stunnel.log'), '%Y-%m-%d %H:%M:%S'),
+        ('celery', os.path.join(env.log_dir, 'celerycam.log'), '%Y-%m-%d %H:%M:%S'),
+        ('celery', os.path.join(env.log_dir, 'celerybeat.log'), '%Y-%m-%d %H:%M:%S'),
+    ] + [('celery', os.path.join(env.log_dir, 'celeryd-%s.log' % w), '%Y-%m-%d %H:%M:%S')
```
Comment (rebecca-caktus) [inline fabulaws/library/wsgiautoscale/api.py]: Why is this last list item separated out?

File: fabulaws/library/wsgiautoscale/api.py [hunk 2]
```diff
@@ -187,12 +187,32 @@ def _setup_env(deployment_tag=None, environment=None, override_servers={}):
     for i, db in enumerate(env.all_databases):
         db.stunnel_port = 7432 + i
     env.db_settings = env.get('db_settings', {})
+    env.setdefault('gelf_log_host', False)
+    if hasattr(env, 'log_host'):
+        print(red("Warning: Setting log_host is deprecated. Set gelf_log_host "
+                  "instead".format(environment=env.environment),
+              bold=True))
+        env.gelf_log_host = env.log_host
```
Comment (rebecca-caktus) [inline fabulaws/library/wsgiautoscale/api.py]: Should `env.gelf_log_host` be preferred if it is already set?

File: fabulaws/library/wsgiautoscale/api.py (hunk 1 above)
Comment (rebecca-caktus) [inline fabulaws/library/wsgiautoscale/api.py]: Oh, I see now. Carry on.

File: fabulaws/library/wsgiautoscale/api.py [hunk 3]
```diff
@@ -1322,7 +1348,8 @@ def mount_encrypted(drive_letter='f'):
         supervisor('start', 'pgbouncer')
         supervisor('start', 'celery')
     # make sure logstash is running after /secure is mounted
-    sudo('service logstash-agent restart')
+    if env.gelf_log_host:
+        sudo('service logstash-agent restart')
     if env.syslog_server:
         sudo('service rsyslog restart')
```
Comment (rebecca-caktus) [inline fabulaws/library/wsgiautoscale/api.py]: Is there no comparable action for `awslogs`?

File: fabulaws/library/wsgiautoscale/api.py [hunk 4]
```diff
@@ -187,12 +187,32 @@ def _setup_env(deployment_tag=None, environment=None, override_servers={}):
...
+              bold=True))
+        env.gelf_log_host = env.log_host
     env.setdefault('syslog_server', False)
+    env.setdefault('awslogs_access_key_id', False)
     if 'production_environments' not in env:
         # Backwards compatibility
         env.production_environments = ['production']
     if 'use_basic_auth' not in env:
         env.use_basic_auth = {}
+    env.log_files = [
```
Comment (rebecca-caktus) [inline fabulaws/library/wsgiautoscale/api.py]: Does this need to be customizable by the end user?

File: fabulaws/library/wsgiautoscale/api.py (hunk 3 above)
Comment (tobiasmcnulty) [inline fabulaws/library/wsgiautoscale/api.py]: good catch!

File: fabulaws/library/wsgiautoscale/api.py [hunk 5]
```diff
@@ -201,6 +201,7 @@ def _setup_env(deployment_tag=None, environment=None, override_servers={}):
         env.production_environments = ['production']
     if 'use_basic_auth' not in env:
         env.use_basic_auth = {}
+    env.setdefault('extra_log_files', [])
```
Comment (rebecca-caktus) [inline fabulaws/library/wsgiautoscale/api.py]: Should the default be a dictionary?

**Comment by rebecca-caktus** [comment]
:rocket:

**Pull request merged**